### PR TITLE
Build for all platforms via FFMPEGBuilder

### PR DESCRIPTION
**Pull request by ianshmean**
Cross-platform FFMPEG build through https://github.com/JuliaIO/FFMPEGBuilder
Test with:
```
] add VideoIO#ffmpegbuilder
] test VideoIO
```

Test status:
# glibc Linuces
- [x] Linux(:i686)
- [x] Linux(:x86_64)
- [x] Linux(:aarch64) - Functional but frame comparison precision failure (CI via drone.io)
- [x] Linux(:armv7l) - Functional but frame comparison precision failure (CI via drone.io)
- [ ] Linux(:powerpc64le)
# musl Linuces
- [ ] Linux(:i686, :musl)
- [ ] Linux(:x86_64, :musl)
- [ ] Linux(:aarch64, :musl)
- [ ] Linux(:armv7l, :musl)
# BSDs
- [x] MacOS(:x86_64)
- [ ] FreeBSD(:x86_64) - Fails to build.. `libavcodec` not satisfied
# Windows
- [x] Windows(:i686)
- [x] Windows(:x86_64)

It would be great to test as much of this list as possible before merging!
Please feel free to reply with success or error messages!

cc. @giordano

### CODE REVIEW

File: deps/buildmod_LibVPX.v1.8.0.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
+
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
+const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
+products = [
```
Comment (giordano) [inline deps/buildmod_LibVPX.v1.8.0.jl]: You could conditionally set it to the proper value or to the empty one, depending on current operating system (with iswindows). You could also just upload this to libvpx release

**Comment by ianshmean** [comment]
VideoIO test status:
# glibc Linuces
- [ ]         Linux(:i686)
- [x]         Linux(:x86_64) - Confirmed locally & Travis
- [ ]         Linux(:aarch64)
- [ ]         Linux(:armv7l)
- [ ]         Linux(:powerpc64le)
# musl Linuces
- [ ]         Linux(:i686, :musl)
- [ ]         Linux(:x86_64, :musl)
- [ ]         Linux(:aarch64, :musl)
- [ ]         Linux(:armv7l, :musl)
# BSDs
- [x]         MacOS(:x86_64) - Confirmed locally & Travis
- [ ]         FreeBSD(:x86_64)
# Windows
- [ ]         Windows(:i686) - Failing on appveyor
- [ ]         Windows(:x86_64) - Failing locally + appveyor

File: deps/buildmod_LibVPX.v1.8.0.jl (hunk 1 above)
Comment (ianshmean) [reply deps/buildmod_LibVPX.v1.8.0.jl]: Good idea! And yeah, the buildmod files were temporary. I just uploaded it to the release

**Comment by ianshmean** [comment]
I've added arm & aarch64 CI via drone.io.

## Status
Precision error on frame comparison for:
- arm & julia 1.0
- aarch64 & julia 1.0
- aarch64 & julia 1.1

Except for:
- arm & julia 1.1 - no v1.1 build image for arm

File: src/VideoIO.jl [hunk 2]
```diff
@@ -5,6 +5,15 @@ using FixedPointNumbers, ColorTypes, ImageCore, Requires, Dates
 
 libpath = joinpath(@__DIR__, "..", "deps", "usr", "lib")
 
+if Sys.iswindows()
+    execenv = ("PATH" => string(VideoIO.libpath,";",Sys.BINDIR))
```
[the remark this answers is not in the document]
Comment (ianshmean) [reply src/VideoIO.jl]: Good point. done

**Comment by ianshmean** [comment]
I'm going to merge as-is:
Only FreeBSD fails to build. 
ARM has precision errors that needs looking into, so the frame comparison tests have been set to `@test_skip` on ARM which get reported as "broken" but don't report as failures on the `drone.io`. Both are explained in `README.md`. 
Full testing status up to date above

Big thanks for the final push to @giordano and @staticfloat! `BinaryBuilder.jl` FTW

**Comment by ianshmean** [comment]
# Update
As of #180 platform test status is:
## glibc Linuces
- [x] Linux(:i686)
- [x] Linux(:x86_64)
- [x] Linux(:aarch64) - Functional but frame comparison precision failure (CI via drone.io)
- [x] Linux(:armv7l) - Functional but frame comparison precision failure (CI via drone.io)
- [ ] Linux(:powerpc64le)
## musl Linuces
- [ ] Linux(:i686, :musl)
- [ ] Linux(:x86_64, :musl)
- [ ] Linux(:aarch64, :musl)
- [ ] Linux(:armv7l, :musl)
## BSDs
- [x] MacOS(:x86_64)
- [x] FreeBSD(:x86_64)
## Windows
- [x] Windows(:i686)
- [x] Windows(:x86_64)

**Pull request merged**